server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -2661,7 +2661,7 @@ protected DataStoreMetadataUpdateResult updateDataSourceMetadataWithHandle(
 
     if (startMetadataGreaterThanExisting && !startMetadataMatchesExisting) {
       // Offsets stored in startMetadata is greater than the last commited metadata.
-      return DataStoreMetadataUpdateResult.failure(
+      return DataStoreMetadataUpdateResult.retryableFailure(
```

Comment:
Please add another line to the comment above clarifying why we should retry in this case.